### PR TITLE
dockerfile update: use debian base image and haproxy run as non root …

### PR DESCRIPTION
**Pull request by othmane399**
…user

### CODE REVIEW

File: 1.9/Dockerfile [hunk 1]
```diff
@@ -52,12 +55,30 @@ RUN set -x \
...
+USER ${USERNAME}
+
 # https://www.haproxy.org/download/1.8/doc/management.txt
 # "4. Stopping and restarting HAProxy"
 # "when the SIGTERM signal is sent to the haproxy process, it immediately quits and all established connections are closed"
 # "graceful stop is triggered when the SIGUSR1 signal is sent to the haproxy process"
 STOPSIGNAL SIGUSR1
 
 COPY docker-entrypoint.sh /
+# username need to be hardcoded
```
Comment (bpaquet) [inline 1.9/Dockerfile]: Please use ${USERNAME}

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1 @@
+1.9/Dockerfile
```
Comment (bpaquet) [inline Dockerfile]: Remove for now

File: docker-entrypoint.sh [hunk 3]
```diff
@@ -0,0 +1 @@
+1.9/docker-entrypoint.sh
```
Comment (bpaquet) [inline docker-entrypoint.sh]: Remove

File: 1.9/Dockerfile [hunk 4]
```diff
@@ -59,15 +59,16 @@ RUN set -x \
...
 
 RUN apt-get update && apt-get install liblua5.3-dev libpcre3-dev libssl-dev zlib1g-dev -y && apt-get clean
 
 ARG USERNAME=haproxy
 
 RUN addgroup ${USERNAME} && useradd --shell /sbin/nologin --create-home -g ${USERNAME} ${USERNAME} && \
 		mkdir -p /usr/local/etc/haproxy && \
-		chown ${USERNAME}:${USERNAME} /usr/local/etc/haproxy
-
+		chown -R ${USERNAME}:${USERNAME} /usr/local/etc/haproxy
```
Comment (bpaquet) [inline 1.9/Dockerfile]: Why do you want to do the chown ?

Comment (othmane399) [reply 1.9/Dockerfile]: to have permissions on start and config files

Comment (bpaquet) [reply 1.9/Dockerfile]: Chmod 755 is enough, no need of chow ?

Comment (bpaquet) [reply 1.9/Dockerfile]: The user does not need to own the config.

Comment (othmane399) [reply 1.9/Dockerfile]: yess t'as raison !

**Pull request merged**